### PR TITLE
Make key repeat configurable

### PR DESCRIPTION
**Pull request by RyanDwyer**
This creates two input commands for configuring the repeat delay and rate.

    input "myidentifier" {
        repeat_delay 250
        repeat_rate 25
    }

Tests (use DRM backend):

* Start sway with variations of the above config
* swaymsg input myidentifier repeat_delay 1000
* swaymsg input myidentifier repeat_rate 10

<s>Attempting to configure it via IPC doesn't work for some reason. It passes the correct values to wlr_keyboard_set_repeat_info() but it doesn't take effect. Is there anything else I need to do or is that a bug in wlroots?</s> (EDIT: This is because I was testing via the X11 backend)

### CODE REVIEW

File: sway/commands/input/repeat.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	int repeat_rate = atoi(argv[0]);
+	if (repeat_rate < 0) {
+		return cmd_results_new(CMD_INVALID, "repeat_rate",
+			"Repeat rate cannot be negative");
+	}
+	new_config->repeat_rate = repeat_rate;
+
+	apply_input_config(new_config);
+	return cmd_results_new(CMD_SUCCESS, NULL, NULL);
+}
```
Comment (SirCmpwn) [inline sway/commands/input/repeat.c]: Split this into two files

**Comment by SirCmpwn** [comment]
>Is there anything else I need to do or is that a bug in wlroots?

Sounds like bug in wlroots maybe

File: sway/commands/input/repeat.c (hunk 1 above)
Comment (RyanDwyer) [reply sway/commands/input/repeat.c]: Resolved in 9d3739a.

**Comment by RyanDwyer** [comment]
I've discovered there's actually nothing wrong with changing the repeat settings via swaymsg. I'm ashamed to admit I'm still using i3 (I have rotated outputs) and am doing most of my testing by using the X11 backend. I think X was controlling the repeat settings that it sent to my sway window. Changing the repeat settings via swaymsg using the drm backend works fine.

**Comment by SirCmpwn** [comment]
>I'm ashamed to admit I'm still using i3 (I have rotated outputs)

Try the alpha!

**Comment by emersion** [comment]
Thanks!

**Pull request merged**